docs/references/debugging-sdk-tools.md

```diff
@@ -0,0 +1,82 @@
+
+## 2. After installing `.yxi`
+After installing the `.yxi`, there will be a new folder created that can be found in `~/%APPDATA%/Roaming/Alteryx/Tools/<TOOL_NAME>` directory for user installs. (For admin installs, please check your Alteryx system installation folder) 
+
+In the created tool folder, navigate to `/site-packages/ayx_plugins/<tool_name>.py`. When opening this file, you should expect to see the exact same code as the plugin code in your tool workspace (`~/your_tool_workspace/backend/ayx_plugins/<tool_name>.py`). To reiterate, when you make some changes to your code in your workspace, recreate and install the `.yxi`, you should expect the same changes to also propogate to this file in the installed tool folder. These checks will confirm that the installation process is working correctly.
+
+## 3. After running in Designer
+If, after you've run the plugin in Designer, the tool does not yield expected results, you can check the logs following this method:
+
+Checking for PythonSDK.log and enabling Debug logging:
+
+- Tool logs are written to a file called PythonSDK.log located in Local AppData. For example, `%LOCALAPPDATA%\Alteryx\Log\PythonSDK.log`.
+
+- Ensure you launch the plugin with the environment variable 
+`AYX_SDK_VERBOSE=1`.
+For example, one way to do this is in the Windows Environment dialog. Windows 10+ users, go to [How Do I Set System Environment Variables in Windows 10](https://superuser.com/questions/949560/how-do-i-set-system-environment-variables-in-windows-10_E) for more information.
+
+For more, please see the [FAQ](https://github.com/alteryx/ayx-developer-sdk/blob/main/docs/references/faq.md#where-are-the-debug-logs-located)
+
+By checking the logs, you will see what is happening during execution time and be able to check your expectations of what you think should be happening with what is actually happening. 
+
+
+## 4. Running the plugin at command line
+If you see an error message that looks like `Internal error: Failed to read port assignment`, you can try to run the `main.pyz` from your tool workspace using the command line to diagnose the error that's preventing the port assignment. Navigate to your tool workspace and run the following command:
+
+`python main.pyz start-sdk-tool-service ayx_plugins <tool_name>`
+A successful run will look like this:
+
+```sh
+(dev) orion.ou ~\tool_workspace [11:54AM]> python .\main.pyz start-sdk-tool-service ayx_plugins Passthrough
+ListenPort: None
+INFO: Starting shivRemoval tool with AMP Provider.
+ListenPort: 62136
+```
+
+If there are any issues with your python code, it would look something like this:
```

Comment:
TW review:

If there are any issues with your Python code, it looks something like this: